src/core/Akka.Persistence.TCK/Query/AllEventsSpec.cs

```diff
@@ -0,0 +1,134 @@
+            var a = Sys.ActorOf(Query.TestActor.Props("a"));
+            var b = Sys.ActorOf(Query.TestActor.Props("b"));
+
+            a.Tell("hello");
+            ExpectMsg("hello-done");
+            a.Tell("world");
+            ExpectMsg("world-done");
+            b.Tell("test");
+            ExpectMsg("test-done");
+
+            var eventSrc = queries.AllEvents(NoOffset.Instance);
+            var probe = eventSrc.RunWith(this.SinkProbe<EventEnvelope>(), Materializer);
+
+            probe.Request(2);
+            probe.ExpectNext<EventEnvelope>(p => p.PersistenceId == "a" && p.SequenceNr == 1L && p.Event.Equals("hello"));
+            probe.ExpectNext<EventEnvelope>(p => p.PersistenceId == "a" && p.SequenceNr == 2L && p.Event.Equals("world"));
+
+            probe.ExpectNoMsg(TimeSpan.FromMilliseconds(500));
+
+            probe.Request(2);
+            probe.ExpectNext<EventEnvelope>(p => p.PersistenceId == "b" && p.SequenceNr == 1L && p.Event.Equals("test"));
+            probe.ExpectComplete();
+        }
+
+        [Fact]
+        public virtual void ReadJournal_query_AllEvents_should_complete_when_no_events()
+        {
+            var queries = ReadJournal as IAllEventsQuery;
+
+            var eventSrc = queries.AllEvents(NoOffset.Instance);
+            var probe = eventSrc.RunWith(this.SinkProbe<EventEnvelope>(), Materializer);
+            probe.Request(2).ExpectComplete();
+        }
+
+        [Fact]
+        public virtual void ReadJournal_query_AllEvents_should_find_events_from_offset_exclusive()
+        {
+            var queries = ReadJournal as IAllEventsQuery;
+
+            var a = Sys.ActorOf(Query.TestActor.Props("a"));
+            var b = Sys.ActorOf(Query.TestActor.Props("b"));
+
+            a.Tell("hello");
+            ExpectMsg("hello-done");
+            a.Tell("world");
+            ExpectMsg("world-done");
+            b.Tell("test");
+            ExpectMsg("test-done");
+
+            var eventSrc1 = queries.AllEvents(NoOffset.Instance);
+            var probe1 = eventSrc1.RunWith(this.SinkProbe<EventEnvelope>(), Materializer);
+            probe1.Request(2);
+            probe1.ExpectNext<EventEnvelope>(p => p.PersistenceId == "a" && p.SequenceNr == 1L && p.Event.Equals("hello"));
+            var offs = probe1.ExpectNext<EventEnvelope>(p => p.PersistenceId == "a" && p.SequenceNr == 2L && p.Event.Equals("world")).Offset;
+            probe1.Cancel();
+
+            var eventSrc2 = queries.AllEvents(offs);
+            var probe2 = eventSrc2.RunWith(this.SinkProbe<EventEnvelope>(), Materializer);
+            probe2.Request(10);
+            // hello and world is not included, since exclusive offset
+            probe2.ExpectNext<EventEnvelope>(p => p.PersistenceId == "b" && p.SequenceNr == 1L && p.Event.Equals("test"));
+            probe2.Cancel();
+        }
+
+        [Fact]
+        public virtual void ReadJournal_query_AllEvents_should_see_all_150_events()
```

Comment:
Is 150 below the pagination threshold?

Reply:
No, the command uses the max buffer size as the pagination size, it is set to 3 in the test.

Reply:
Ok good, so just so I understand correctly - this spec forces the journal to work through several pages worth of results in separate queries, right?

Reply:
yes, it retrieves the events through separate queries, using "WHERE ordering > lastindex" and "LIMIT x"